src/shared/sam/cli/samCliBuild.ts

```diff
@@ -125,21 +127,40 @@ export class SamCliBuildInvocation {
                     'AWS.sam.build.failure.diskSpace',
                     '"sam build" failed. Check system disk space.'
                 )
+            } else if (timer !== undefined) {
+                timer.refresh()
             }
         }
 
-        const childProcessResult = await this.args.invoker.invoke({
+        let childProcessResult: Promise<ChildProcessResult | void> = this.args.invoker.invoke({
             spawnOptions: { env },
             arguments: invokeArgs,
             onStdout: checkFailure,
             onStderr: checkFailure,
```

Comment:
guess the handler should be renamed
```suggestion
            onStderr: onOutput,
```